lib/internal/streams/operators.js

```diff
@@ -250,7 +235,7 @@ function filter(fn, options) {
     }
     return kEmpty;
   }
-  return this.map(filterFn, options);
+  return map.call(this, filterFn, options);
```

Comment:
it would be great to have a test for this (not calling `map` if it was over-riden on the stream)

Reply:
added to other operators that call `map` as well